sensors/depth/depth.go

```diff
@@ -0,0 +1,53 @@
+// Package depth implements the Depth sensor
+package depth
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"go.viam.com/rdk/components/camera"
+	"go.viam.com/rdk/registry"
+
+	"go.viam.com/slam/config"
+	"go.viam.com/slam/sensors"
+	"go.viam.com/slam/sensors/utils"
+)
+
+// Depth represents a depth sensor.
+type Depth struct {
+	Name         string
+	depth        camera.Camera
+	DataRateMsec int
+}
+
+// New creates a new Depth sensor based on the sensor definition and the service config.
+func New(ctx context.Context, deps registry.Dependencies, sensor sensors.Sensor, svcConfig *config.AttrConfig) (Depth, error) {
+	name, err := sensor.GetName(svcConfig)
+	if err != nil {
+		return Depth{}, err
+	}
+
+	newDepth, err := camera.FromDependencies(deps, name)
+	if err != nil {
+		return Depth{}, errors.Wrapf(err, "error getting camera %v for slam service", name)
+	}
+
+	if err = validate(ctx, newDepth); err != nil {
+		return Depth{}, err
+	}
+
+	return Depth{
+		Name:         name,
+		depth:        newDepth,
+		DataRateMsec: sensor.GetDataRateMsec(svcConfig),
+	}, nil
+}
+
+// GetData returns data from the depth sensor.
```

Comment:
Do we want to include more information about this very complex type signature? From the type signature alone, it is not clear what the func() return value is. Yet, it is important that the user releases the image.

Reply:
`func()` is only only `!= nil` if `error == nil`. I'll add a comment, though, to clarify this.

Reply:
But what is `func()`? From this type signature and comment alone, I have no idea.

Reply:
It is a release function you must call once you're done using the image. As of Sept 27, 2022, it didn't do anything yet. I am not sure if that has changed since then or where this function is defined. @edaniels might know more.